libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -122,8 +122,8 @@ struct virq_handle {
 
 typedef struct virq_handle *virq_handle_t;
 
-/* Inject interrupt into vcpu */
-static int vgic_vcpu_inject_irq(struct vgic_dist_device *d, vm_vcpu_t *inject_vcpu, struct virq_handle *irq);
+/* Load IRQ into hardware list register */
+static int vgic_vcpu_load_lr(struct vgic_dist_device *d, vm_vcpu_t *vcpu, struct virq_handle *irq);
```

Comment:
I thing we don't need this prototype here, the function is only called after the implementation.

Reply:
ah no, we need it

Reply:
or well, just change the order in the cleanup and we can drop this prototype and get rid of some redundancy in the code

Reply:
Done.